### PR TITLE
Update docs for devices flag

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Fixes #\<issue_number>

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/common/trainer.rst [hunk 1]
```diff
@@ -543,6 +543,37 @@ will need to be set up to use remote filepaths.
...
+
+    # Training with TPU Accelerator using 8 tpu cores
+    trainer = Trainer(devices=8, accelerator="tpu")
+
+.. note:: The ``"auto"`` option recognizes the devices to train on, depending on the ``Accelerator`` being used.
+
+.. code-block:: python
+
+    # If your machine has GPUs, it will use all the available GPUs for training
+    trainer = Trainer(devices="auto", accelerator="auto")
```
Comment (rohitgr7) [inline docs/source/common/trainer.rst:569]: for CPU?

File: docs/source/common/trainer.rst [hunk 2]
```diff
@@ -1176,7 +1207,7 @@ Half precision, or mixed precision, is the combined use of 32 and 16 bit floatin
 
         pip install --no-cache-dir --global-option="--cpp_ext" --global-option="--cuda_ext" https://github.com/NVIDIA/apex
 
-    2. Set the `precision` trainer flag to 16. You can customize the `Apex optimization level <https://nvidia.github.io/apex/amp.html#opt-levels>`_ by setting the `amp_level` flag.
+    1. Set the `precision` trainer flag to 16. You can customize the `Apex optimization level <https://nvidia.github.io/apex/amp.html#opt-levels>`_ by setting the `amp_level` flag.
```
Comment (rohitgr7) [inline docs/source/common/trainer.rst:1210]: first one is `Install Apex` up there.
```suggestion
    2. Set the ``precision`` trainer flag to 16. You can customize the `Apex optimization level <https://nvidia.github.io/apex/amp.html#opt-levels>`_ by setting the `amp_level` flag.
```

**Review by rohitgr7 — APPROVED** [review]

File: docs/source/common/trainer.rst (hunk 1 above)
Comment (kaushikb11) [reply docs/source/common/trainer.rst:569]: Updated!

**Review by tchaton — APPROVED** [review]
LGMT !

**Review by ethanwharris — APPROVED** [review]

File: docs/source/common/trainer.rst [hunk 3]
```diff
@@ -543,6 +543,40 @@ will need to be set up to use remote filepaths.
...
+    # Training with CPU Accelerator using 2 processes
+    trainer = Trainer(devices=2, accelerator="cpu")
+
+    # Training with GPU Accelerator using GPUs 1 and 3
+    trainer = Trainer(devices=[1, 3], accelerator="gpu")
+
+    # Training with TPU Accelerator using 8 tpu cores
+    trainer = Trainer(devices=8, accelerator="tpu")
+
+.. note:: The ``"auto"`` option recognizes the devices to train on, depending on the ``Accelerator`` being used.
```
Comment (Borda) [inline docs/source/common/trainer.rst:564]:
```suggestion
.. tip:: The ``"auto"`` option recognizes the devices to train on, depending on the ``Accelerator`` being used.
```

**Review by Borda — APPROVED** [review]
